### DISCUSSION TITLE
Roadmap Q4 2022

### DISCUSSION THREAD

**Discussion by msujew** [eclipse-langium/langium#718] [Announcements]
Here's a list of what the contributors of Langium plan to do for the next 3 months, with some more probably still to come. At the end of Q4 we will likely have our first full release of Langium (1.0.0)

- [x] @montymxb @Lotes @pluralia Improve type inference and validation using formal methods
- [ ] @msujew @Lotes Model serialization (stringification) (moved to Q1/Q2 of 2023)
- [x] @msujew ANTLR4 style unbounded lookahead https://github.com/langium/langium/pull/755

**Comment by spoenemann** [comment]
Hopefully the next release will include https://github.com/Chevrotain/chevrotain/pull/1852 as well.

**Reply by luan-xiaokun** [reply]
If I understand this new feature right, chevrotain will support LL(*) parsing, which is the same as ANTLR. So will the  next release support writing left-recursive parse rules like ANTLR?

**Reply by spoenemann** [reply]
No, left-recursive rules are not covered and would need another solution. LL(*) parsing covers grammars with unbounded lookahead, so we don't need to know the number of lookahead tokens in advance.
